cloudflare/resource_cloudflare_load_balancer_monitor.go

```diff
@@ -0,0 +1,261 @@
+	}
+
+	log.Printf("[DEBUG] Update CloudFlare Load Balancer Monitor from struct: %+v", loadBalancerMonitor)
+
+	_, err := client.ModifyLoadBalancerMonitor(loadBalancerMonitor)
+	if err != nil {
+		return errors.Wrap(err, "error modifying load balancer monitor")
+	}
+
+	log.Printf("[INFO] CloudFlare Load Balancer Monitor %q was modified", d.Id())
+
+	return resourceCloudFlareLoadBalancerPoolMonitorRead(d, meta)
+}
+
+func expandLoadBalancerMonitorHeader(cfgSet interface{}) map[string][]string {
+	header := make(map[string][]string)
+	cfgList := cfgSet.(*schema.Set).List()
+	for _, item := range cfgList {
+		cfg := item.(map[string]interface{})
+		header[cfg["header"].(string)] = expandInterfaceToStringList(cfg["values"].(*schema.Set).List())
+	}
+	return header
+}
+
+func resourceCloudFlareLoadBalancerPoolMonitorRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	loadBalancerMonitor, err := client.LoadBalancerMonitorDetails(d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "HTTP status 404") {
+			log.Printf("[INFO] Load balancer monitor %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		} else {
+			return errors.Wrap(err,
+				fmt.Sprintf("Error reading load balancer monitor from API for resource %s ", d.Id()))
+		}
+	}
+	log.Printf("[DEBUG] Read CloudFlare Load Balancer Monitor from API as struct: %+v", loadBalancerMonitor)
+
+	d.Set("expected_body", loadBalancerMonitor.ExpectedBody)
+	d.Set("expected_codes", loadBalancerMonitor.ExpectedCodes)
+	d.Set("method", loadBalancerMonitor.Method)
+	d.Set("timeout", loadBalancerMonitor.Timeout)
+	d.Set("path", loadBalancerMonitor.Path)
+	d.Set("interval", loadBalancerMonitor.Interval)
+	d.Set("retries", loadBalancerMonitor.Retries)
+	d.Set("type", loadBalancerMonitor.Type)
+	d.Set("description", loadBalancerMonitor.Description)
+	d.Set("created_on", loadBalancerMonitor.CreatedOn.Format(time.RFC3339Nano))
+	d.Set("modified_on", loadBalancerMonitor.ModifiedOn.Format(time.RFC3339Nano))
+
+	if err := d.Set("header", flattenLoadBalancerMonitorHeader(loadBalancerMonitor.Header)); err != nil {
+		log.Printf("[WARN] Error setting header for load balancer monitor %q: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func flattenLoadBalancerMonitorHeader(header map[string][]string) *schema.Set {
+	flattened := make([]interface{}, 0)
+	for k, v := range header {
+		cfg := map[string]interface{}{
+			"header": k,
+			"values": schema.NewSet(schema.HashString, flattenStringList(v)),
+		}
+		flattened = append(flattened, cfg)
+	}
+	return schema.NewSet(HashByMapKey("header"), flattened)
+}
+
+func resourceCloudFlareLoadBalancerPoolMonitorDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	log.Printf("[INFO] Deleting CloudFlare Load Balancer Monitor: %s ", d.Id())
+
+	err := client.DeleteLoadBalancerMonitor(d.Id())
+	if err != nil {
+		return errors.Wrap(err, "error deleting cloudflare load balancer monitor")
```

Comment:
Check the error here for the equivalent of a 404 like it's done on line 208 above. If it's delete externally to Terraform, we don't want users to be stuck here.

Reply:
Is this in case deletions occur concurrently with terraform run? I would assume that otherwise terraform will always run read first. In any case I've addressed this now

Reply:
It likely `READ` will be ran first, but it's plausible that we're executing a saved plan in which case I believe it won't